keylime/migrations/versions/f4196d13fe45_migrate_mb_refstates_to_mbpolicies.py

```diff
@@ -0,0 +1,83 @@
+"""migrate mb_refstates to mbpolicies
+
+Revision ID: f4196d13fe45
+Revises: 32902c0a8d90
+Create Date: 2024-02-08 04:16:21.554062
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "f4196d13fe45"
+down_revision = "32902c0a8d90"
+branch_labels = None
+depends_on = None
+
+
+def upgrade(engine_name):
+    globals()[f"upgrade_{engine_name}"]()
+
+
+def downgrade(engine_name):
+    globals()[f"downgrade_{engine_name}"]()
+
+
+def upgrade_registrar():
+    pass
+
+
+def downgrade_registrar():
+    pass
+
+
+def upgrade_cloud_verifier():
+    # Migrate existing agent info to the mbpolicies table.
+    conn = op.get_bind()
+
+    res = conn.execute(sa.text("SELECT agent_id, mb_refstate FROM verifiermain"))
+    results = res.fetchall()
+    old_policy = [{"name": r[0], "mb_policy": r[1]} for r in results]
+
+    # get existing table metadata
+    meta = sa.MetaData()
+    meta.reflect(bind=conn, only=("mbpolicies",))
+    mbpolicies = meta.tables["mbpolicies"]
+
+    # Insert the existing MB policies to the mbpolicies table
+    op.bulk_insert(mbpolicies, old_policy)
+
+    # Modify verifiermain table (i.e. add mb_policy_id and remove mb_refstate)
+    with op.batch_alter_table("verifiermain") as batch_op:
+        batch_op.add_column(
+            sa.Column("mb_policy_id", sa.Integer(), sa.ForeignKey("mbpolicies.id", name="fk_verifiermain_mbpolicies"))
```

Comment:
When I tried downgrading I found that the CONSTRAINT still remained in `verifiermain`:
```
CONSTRAINT fk_verifiermain_allowlists FOREIGN KEY(ima_policy_id) REFERENCES allowlists (id)
```

Reply:
I used this command here to downgarde:
```
sudo bash -c "PYTHONPATH=. alembic -c keylime/migrations/alembic.ini downgrade 160f932fde5b"
```

Reply:
> When I tried downgrading I found that the CONSTRAINT still remained in `verifiermain`:
> 
> ```
> CONSTRAINT fk_verifiermain_allowlists FOREIGN KEY(ima_policy_id) REFERENCES allowlists (id)
> ```
> 
> Can you drop this CONSTRAINT during a downgrade?

That constraint is beacuse of the existing `allowlists` table and it should not be removed when we downgrade till 160f932fde5b .

Reply:
> That constraint is beacuse of the existing `allowlists` table and it should not be removed when we downgrade till 160f932fde5b .

I downgraded to `160f932fde5b` with this command:

```
sudo bash -c "PYTHONPATH=. alembic -c keylime/migrations/alembic.ini downgrade 160f932fde5b
```

Reply:
@niteeshkd You're right. The constraint is gone.